rubocop/missing_keys_on_shared_area.rb

```diff
@@ -0,0 +1,82 @@
+require 'rubocop'
+
+module RuboCop
+  module Lint
+    class MissingKeysOnSharedArea < RuboCop::Cop::Cop
+      MISSING_KEYS_MSG = "There are missing keys on the shared area. Check constants provided in 'SharedValues' or keys provided in 'output' method in the action's code".freeze
+      MISSING_OUTPUT_METHOD_MSG = "There are declared keys on the shared area 'SharedValues', but 'output' method has not been found".freeze
+
+      attr_writer :shared_values_constants
+      def shared_values_constants
+        @shared_values_constants ||= []
+      end
+
+      def on_module(node)
```

Comment:
I believe we could simplify the logic of getting constants defined in `SharedModule` module by using `def on_const(node)` however I could get a module name for the constants there. Does anyone have an idea how to do that? 🤔

Reply:
We may need to call upon @milch for help on that? 🙃 🙏

Reply:
If you want to find all constant assignments (which seems to be what you are doing looking at the `casgn` check) under the module node you should be able to define a [node searcher](https://www.rubydoc.info/gems/rubocop/RuboCop/NodePattern/Macros#def_node_search-instance_method) similar to this: 

```ruby
def_node_search :extract_const_assignment, <<-PATTERN
  (casgn nil? $_ ...)
PATTERN
 ```

It’s been a while since I’ve written a RuboCop though so might be a bit rusty here with the syntax.

Reply:
Thank you @milch 😍that works very well, I changed the code!